magma/backend/mlir/mlir_printer_utils.py

```diff
@@ -46,6 +46,14 @@ def print_signature(
     printer.print(", ".join(signatures))
 
 
+def _attr_to_string(value: Any) -> str:
+    if isinstance(value, MlirAttribute):
+        return value.emit()
```

Comment:
Could we simply define `__str__` on `MlirAttribute` instead?

Reply:
We could, but it is nice to have the str/repr inherited from dataclass since most attributes are defined as such. This helps with debugging.

The right thing to do here would be to wrap POD-type attributes in special classes, but I haven't done that yet.

Reply:
See #1224, will rebase